middleware/hooks.go

```diff
@@ -167,9 +167,15 @@ func (h *responseHook) appendPeers(req *bittorrent.AnnounceRequest, resp *bittor
 
 	switch req.IP.AddressFamily {
 	case bittorrent.IPv4:
-		resp.IPv4Peers = peers
+		resp.IPv4Peers = append(resp.IPv4Peers, peers...)
```

Comment:
We could definitely do this instead ~L152:

```go
numWant := req.NumWant - len(resp.IPv4Peers)
if numWant > 0 {
  peers, err := h.store.AnnouncePeers(req.InfoHash, seeding, numWant, req.Peer)
}
...
```
This way, we can exit early if we don't need anymore peers.